### PR TITLE
refactor(cli/repl): tightly integrate event loop

### PR DESCRIPTION
**Pull request by caspervonb**
This tightly integrates the event loop into the repl event loop for more control over it so it doesn't have to spin wildly outside of the inner loop for us to get the inspector to pump.

### CODE REVIEW

File: cli/repl.rs [hunk 1]
```diff
@@ -29,18 +32,70 @@ impl Validator for Helper {
   }
 }
 
+async fn post_message_and_poll(
+  session: &mut InspectorSession,
+  method: String,
```
Comment (bartlomieju) [inline cli/repl.rs:37]: nitpick: use `&str`

File: cli/repl.rs [hunk 2]
```diff
@@ -29,18 +32,70 @@ impl Validator for Helper {
...
+  params: Option<Value>,
+  worker: &mut Worker,
```
Comment (bartlomieju) [inline cli/repl.rs:39]: nitpick, make it first or second argument

Comment (caspervonb) [reply cli/repl.rs:39]: All the arguments are selected here but I'll assume you mean the worker 🙃

Comment (nayeemrmn) [reply cli/repl.rs:39]: It's a (bottom) one line selection unless it says `Comments on line x to y`.

File: cli/repl.rs [hunk 3]
```diff
@@ -29,17 +32,64 @@ impl Validator for Helper {
...
+  let mut line =
+    tokio::task::spawn_blocking(move || editor.lock().unwrap().readline("> "));
+
+  loop {
+    tokio::select! {
+      result = &mut line => {
+        return result.unwrap();
+      }
+      _ = &mut *worker => {
+        tokio::time::delay_for(tokio::time::Duration::from_millis(0)).await;
```
Comment (caspervonb) [inline cli/repl.rs:70]: Just returning `line.await` worked great locally but completely locks up the CI with no test output at all.

File: cli/repl.rs [hunk 4]
```diff
@@ -29,17 +32,76 @@ impl Validator for Helper {
...
+    tokio::select! {
+      result = &mut response => {
+        return result
+      }
+
+      _ = &mut *worker => {
+        // A zero delay is long enough to yield the thread in order to prevent the loop from
+        // running hot for messages that are taking longer to resolve like for example an
+        // evaluation of top level await.
+        tokio::time::delay_for(tokio::time::Duration::from_millis(0)).await;
```
Comment (bartlomieju) [inline cli/repl.rs:54]: Thanks for the comment

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks for the fix @caspervonb

**Pull request merged**